packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -578,6 +641,99 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     return slicesOfTokensToDetect;
   }
 
+  /**
+   * This adds detected tokens from the Accounts API, avoiding the multi-call RPC calls for balances
+   * @param options - method arguments
+   * @param options.selectedAddress - address to check against
+   * @param options.chainId - chainId to check tokens for
+   * @param options.tokenCandidateSlices - these are tokens we know a user does not have (by checking the tokens controller).
+   * We will use these these token candidates to determine if a token found from the API is valid to be added on the users wallet.
+   * It will also prevent us to adding tokens a user already has
+   * @returns a success or failed object
+   */
+  async #addDetectedTokensViaAPI({
```

Comment:
Hmm this was nice to write, but concerned that it will be tricky to test, there is a lot of setup required (but that is mostly because it uses things inside this class.)

I'll see when writing the tests.

Reply:
OK tests weren't too bad to write.

It was a bit finicky working with the existing test suite - tried to keep the tests expressive.